pkg/process/checks/net.go

```diff
@@ -85,12 +90,102 @@ func (c *ConnectionsCheck) Run(cfg *config.AgentConfig, groupID int32) ([]model.
 	// Resolve the Raddr side of connections for local containers
 	LocalResolver.Resolve(conns)
 
+	// Get the container and process relationship from either the process or container checks
+	ctrsByPid := ctrIDsByPIDs(connectionPIDs(conns.Conns))
+
+	resolveLoopbackConnections(conns.Conns, ctrsByPid)
+
 	tel := c.diffTelemetry(conns.Telemetry)
 
 	log.Debugf("collected connections in %s", time.Since(start))
 	return batchConnections(cfg, groupID, c.enrichConnections(conns.Conns), conns.Dns, c.networkID, tel), nil
 }
 
+func resolveLoopbackConnections(conns []*model.Connection, ctrsByPid map[int32]string) {
+
+	ctrsByLaddr := make(map[model.ContainerAddr]string)
+	ctrsByLoopback := make(map[addrWithNS]string)
+	for _, conn := range conns {
+		cid, ok := ctrsByPid[conn.Pid]
+		if !ok {
+			continue
+		}
+
+		conn.Laddr.ContainerId = cid
+
+		laddr := translatedLaddr(conn.Laddr, conn.IpTranslation)
+		ip := procutil.AddressFromString(laddr.Ip)
+		if ip == nil {
+			continue
+		}
+
+		claddr := model.ContainerAddr{
+			Ip:       laddr.Ip,
+			Port:     laddr.Port,
+			Protocol: conn.Type,
+		}
+
+		if ip.IsLoopback() {
+			ctrsByLoopback[addrWithNS{claddr, conn.NetNS}] = cid
+		} else {
+			ctrsByLaddr[claddr] = cid
+		}
+	}
+
+	log.Tracef("ctrsByLoopback = %v", ctrsByLoopback)
+	log.Tracef("ctrsByLaddr = %v", ctrsByLaddr)
+
+	for _, conn := range conns {
+		if conn.Raddr.ContainerId != "" {
+			log.Tracef("skipping already resolved raddr %v", conn.Raddr)
```

Comment:
Is all this logging useful for development, or should we keep it in the final version?